### ISSUE TITLE
Tls communication fails while using the engine tpm2tss with OpenSSL 3.1.0 

### ISSUE THREAD

**Issue by ehb2hi** [openssl/openssl#20675]
If I try to establish a tls communication with a server using for example this command:

    ```
    $ openssl s_client -connect example.com:8883 -servername example.com -key 0x81000001 -cert client.crt -keyform engine -engine tpm2tss
    ```
I get this error.
    ```
$4057BF5CB17F0000:error:0A00041B:SSL routines:ssl3_read_bytes:tlsv1 alert decrypt error:../ssl/record/rec_layer_s3.c:1584:SSL alert number 51
    ```
I tested also the tools mosquito and curl and I got the same problem. The both tools use also the openssl engine functionality.

Is this issue already known or the openssl3 supports further the usage of the engine and for me specially the tpm2tss engine.  

the tested version is: OpenSSL 3.1.0 14 Mar 2023 (Library: OpenSSL 3.0.2 15 Mar 2022)

**Comment by t8m** [comment]
I'd suggest at least upgrading the library (or try to use LD_LIBRARY_PATH) to the 3.0.8 or 3.1.0 version as there were some fixes related to engines since 3.0.2.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further input
